dialog.cpp

```diff
@@ -0,0 +1,225 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <FL/Fl.H>
+#include <FL/Fl_Button.H>
+#include <FL/Fl_Check_Button.H>
+#include <FL/Fl_Double_Window.H>
+#include <FL/Fl_PNG_Image.H>
+
+#include <algorithm>
+#include <sstream>
+#include <vector>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "desktopenvironments.h"
+#include "dialog_images.h"
+
+const char *msg_launch      = "\nLaunch app\n\n";
+const char *msg_menu        = "\nCreate menu entry\nand launch app";
+const char *msg_checkbox    = " Don't show this message again";
+
+const char *msg_launch_de   = "\nStarte App\n\n";
+const char *msg_menu_de     = "\nMenüeintrag erstellen\nund App starten";
+const char *msg_checkbox_de = " Dieses Fenster nicht erneut anzeigen";
+
+enum {
+  RET_LAUNCH_APP = 0,
+  RET_MENU_ENTRY = 1,
+  RET_DISABLE_MESSAGE = 2,
+  RET_CLOSE_WINDOW = 3
+};
+
+Fl_Double_Window *win;
+int rv = 0;
+bool checkbutton_set = false;
+
+std::vector<std::string> splitString(const std::string &str, const char delimiter = ' ')
+{
+  std::vector<std::string> result;
+  std::stringstream ss(str);
+  std::string item;
+
+  if (str.empty())
+  {
+    ss << " ";
+  }
+
+  while (std::getline(ss, item, delimiter))
+  {
+    result.push_back(item);
+  }
+  return result;
+}
+
+void get_system_font(std::string &font)
+{
+  std::vector<std::string> splitted;
+  std::stringstream ss;
+  int vsize;
+
+  auto *desktopEnvironment = IDesktopEnvironment::getInstance();
+
+  if (desktopEnvironment == nullptr || !(desktopEnvironment->gtkInterfaceFont(font)))
+  {
+    font = "Helvetica";
+    return;
+  }
+
+  splitted = splitString(font);
+  vsize = splitted.size() - 1;
+  ss << atoi(splitted[vsize].c_str());
+
+  /* remove font size */
+  if (ss.str() == splitted[vsize])
+  {
+    splitted.pop_back();
+    vsize = splitted.size() - 1;
+  }
+
+  /* remove font slant */
+  if (splitted[vsize] == "Italic" || splitted[vsize] == "Oblique" ||
+      splitted[vsize] == "Roman") // Times New Roman = Times New??
+  {
+    splitted.pop_back();
+    vsize = splitted.size() - 1;
+  }
+
+  /* remove font weight */
+  if (splitted[vsize] == "Bold" || splitted[vsize] == "Light" ||
+      splitted[vsize] == "Medium" || splitted[vsize] == "Demi-Bold" ||
+      splitted[vsize] == "Black")
+  {
+    splitted.pop_back();
+    vsize = splitted.size() - 1;
+  }
+
+  font.clear();
+
+  for (int i = 0; i < vsize; ++i)
+  {
+    font = font + splitted[i] + " ";
+  }
+  font = font + splitted[vsize];
```

Comment:
That's just redundant, why not use `i <= vsize` above?

That said, I'd rather have used a `stringstream` here, it's slightly more efficient.

Also, you could replace your loop with this nifty construction:

```
std::ostringstream newFont;
std::copy(splitted.begin(), splitted.end(),
           std::ostream_iterator<std::string>(newFont, " "));
font = newFont.str();
```

But that's not really a necessity.